src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -952,14 +954,18 @@ protected static bool IsDefinedInSourceTree(SyntaxReference syntaxRef, SyntaxTre
             => syntaxRef.SyntaxTree == tree &&
                 (!definedWithinSpan.HasValue || syntaxRef.Span.IntersectsWith(definedWithinSpan.Value));
 
-        internal static void ForceCompleteMemberByLocation(SourceLocation locationOpt, Symbol member, CancellationToken cancellationToken)
+#nullable enable
+        internal static void ForceCompleteMemberByLocation(SourceLocation? locationOpt, Predicate<ISymbolInternal>? filter, Symbol member, CancellationToken cancellationToken)
         {
-            if (locationOpt == null || member.IsDefinedInSourceTree(locationOpt.SourceTree, locationOpt.SourceSpan, cancellationToken))
+            if ((locationOpt == null && filter == null)
```

Comment:
>(locationOpt == null && filter == null) [](http://example.com/codeflow?start=16&length=39)

I think this check is unnecessary as a separate check, and the entire 'if' condition is unnecessary complex and confusing. It looks like it can be as simple as:

```
if ((locationOpt == null || member.IsDefinedInSourceTree(locationOpt.SourceTree, locationOpt.SourceSpan, cancellationToken)) &&
(filter == null || filter(member)) )
```